src/stats/stats.jl

```diff
@@ -0,0 +1,165 @@
+export AbstractExecutionStats, GenericExecutionStats,
+       AbstractExtraStats, DictExtraStats,
+       set_stat!, get_stat, statshead, statsline
+
+const STATUSES = Dict(:unknown => "unknown",
+                      :first_order => "first-order stationary",
+                      :max_eval => "maximum number of function evaluations",
+                      :max_time => "maximum elapsed time",
+                      :max_iter => "maximum iteration",
+                      :neg_pred => "negative predicted reduction",
+                      :unbounded => "objective function may be unbounded from below",
+                      :exception => "unhandled exception",
+                      :stalled => "stalled"
+                     )
+
+abstract type AbstractExecutionStats end
+
+abstract type AbstractExtraStats <: AbstractExecutionStats end
+
+type GenericExecutionStats <: AbstractExecutionStats
+  status :: Symbol
+  solution :: Vector # x
+  objective :: Float64 # f(x)
+  dual_feas :: Float64 # ‖∇f(x)‖₂ for unc, ‖P[x - ∇f(x)] - x‖₂ for bnd, etc.
+  iter :: Int
+  counters :: NLPModels.Counters
+  elapsed_time :: Float64
+  solver_specific :: AbstractExtraStats
```

Comment:
On second thought, maybe `solver_specific` should just be a `Dict{Symbol,Any}` and that would simplify things?

Reply:
I think so too.